### PR TITLE
add recipe for dulwich

### PR DESCRIPTION
**Pull request by mikofski**
the pure python git implementation

### CODE REVIEW

File: recipes/dulwich/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (synapticarbors) [inline recipes/dulwich/meta.yaml]: Please delete all comments

File: recipes/dulwich/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  # Uncomment the following line if the package is pure python and the recipe is exactly the same for all platforms.
+  # See https://conda-forge.org/docs/meta.html#building-noarch-packages for more details.
+  # noarch: python
+  number: 1
```
Comment (synapticarbors) [inline recipes/dulwich/meta.yaml]: The `number` should start at `0`. Generally this is only updated in the feedstock if something other than the version of the package changes.

File: recipes/dulwich/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    - dulwich.server
+    - dulwich.walk
+    - dulwich.web
+
+about:
+  home: https://www.dulwich.io/
+  # Remember to specify the license variants for BSD, Apache, GPL, and LGLP.
+  # Prefer the short version, e.g: GPL-2.0 instead of GNU General Public License version 2.0
+  # See https://opensource.org/licenses/alphabetical
+  license: Apache
```
Comment (synapticarbors) [inline recipes/dulwich/meta.yaml]: `license: Apache-2.0`

File: recipes/dulwich/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  # The license_family, i.e. "BSD" if license is "BSD-3-Clause". (optional)
+  license_family: Apache
+  # It is strongly encouraged to include a license file in the package,
+  # (even if the license doesn't require it) using the license_file entry.
+  # See http://conda.pydata.org/docs/building/meta-yaml.html#license-file
+  license_file: CONTRIBUTING.md
```
Comment (synapticarbors) [inline recipes/dulwich/meta.yaml]: It looks like this should be `license_file: COPYING`

File: recipes/dulwich/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    The project is named after the part of London that Mr. and Mrs. Git live in in the particular Monty Python sketch.
+  doc_url: https://www.dulwich.io/docs/
+  dev_url: https://github.com/jelmer/dulwich
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
+    # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
+    - mikofski
+    - jelmer
```
Comment (synapticarbors) [inline recipes/dulwich/meta.yaml]: @jelmer -- are you ok being added as a co-maintainer of the `dulwhich` recipe on conda-forge?

https://conda-forge.org/docs/recipe.html#maintainer-role

**Comment by jelmer** [comment]
Thanks for adding a Dulwich recipe for conda. Unfortunately I don't think I would be a good co-maintainer since I don't use conda myself.

On 10 January 2018 22:23:39 GMT+08:00, Joshua Adelman <notifications@github.com> wrote:
>synapticarbors requested changes on this pull request.
>
>
>
>> @@ -0,0 +1,97 @@
>+# Note: there are many handy hints in comments in this example --
>remove them when you've finalized your recipe
>
>Please delete all comments
>
>> +#  `conda install openssl -c conda-forge``
>+
>+package:
>+  name: {{ name|lower }}
>+  version: {{ version }}
>+
>+source:
>+  fn: {{ name }}-{{ version }}.tar.gz
>+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{
>name }}-{{ version }}.tar.gz
>+  sha256: {{ sha256 }}
>+
>+build:
>+  # Uncomment the following line if the package is pure python and the
>recipe is exactly the same for all platforms.
>+  # See
>https://conda-forge.org/docs/meta.html#building-noarch-packages for
>more details.
>+  # noarch: python
>+  number: 1
>
>The `number` should start at `0`. Generally this is only updated in the
>feedstock if something other than the version of the package changes. 
>
>> +    - dulwich.hooks
>+    - dulwich.ignore
>+    - dulwich.objectspec
>+    - dulwich.patch
>+    - dulwich.reflog
>+    - dulwich.refs
>+    - dulwich.server
>+    - dulwich.walk
>+    - dulwich.web
>+
>+about:
>+  home: https://www.dulwich.io/
>+  # Remember to specify the license variants for BSD, Apache, GPL, and
>LGLP.
>+  # Prefer the short version, e.g: GPL-2.0 instead of GNU General
>Public License version 2.0
>+  # See https://opensource.org/licenses/alphabetical
>+  license: Apache
>
>`license: Apache-2.0`
>
>> +    - dulwich.server
>+    - dulwich.walk
>+    - dulwich.web
>+
>+about:
>+  home: https://www.dulwich.io/
>+  # Remember to specify the license variants for BSD, Apache, GPL, and
>LGLP.
>+  # Prefer the short version, e.g: GPL-2.0 instead of GNU General
>Public License version 2.0
>+  # See https://opensource.org/licenses/alphabetical
>+  license: Apache
>+  # The license_family, i.e. "BSD" if license is "BSD-3-Clause".
>(optional)
>+  license_family: Apache
>+  # It is strongly encouraged to include a license file in the
>package,
>+  # (even if the license doesn't require it) using the license_file
>entry.
>+  # See
>http://conda.pydata.org/docs/building/meta-yaml.html#license-file
>+  license_file: CONTRIBUTING.md
>
>It looks like this should be `license_file: COPYING`
>
>> +  summary: 'Python Git Library'
>+
>+  # The remaining entries in this section are optional, but
>recommended
>+  description: |
>+    Python implementation of the Git file formats and protocols,
>without the need to have git installed.
>+    All functionality is available in pure Python. Optional C
>extensions can be built for improved performance.
>+    The project is named after the part of London that Mr. and Mrs.
>Git live in in the particular Monty Python sketch.
>+  doc_url: https://www.dulwich.io/docs/
>+  dev_url: https://github.com/jelmer/dulwich
>+
>+extra:
>+  recipe-maintainers:
>+    # GitHub IDs for maintainers of the recipe.
>+    # Always check with the people listed below if they are OK
>becoming maintainers of the recipe. (There will be spam!)
>+    - mikofski
>+    - jelmer
>
>@jelmer -- are you ok being added as a co-maintainer of the `dulwhich`
>recipe on conda-forge?
>
>https://conda-forge.org/docs/recipe.html#maintainer-role
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/conda-forge/staged-recipes/pull/4774#pullrequestreview-87846968

-- 
Jelmer Vernooĳ <jelmer@jelmer.uk>
https://www.jelmer.uk/

**Comment by mikofski** [comment]
Thanks @jelmer. I also added #4777 for fastimport. A lot of my colleagues are using Anaconda Python b/c of NumPy and other math and science libraries, so having dulwich available to them is very helpful. Thanks so much for creating dulwich - it is a very valuable piece of software!

**Comment by synapticarbors** [comment]
@mikofski -- Thanks for your contribution to conda-forge. A feedstock for this recipe should be generated soon and you'll be receiving several automated messages about it. When the feedstock has been rendered, please take a look at the README in that repo for information about updating and maintaining the recipe. One of the key points is all updates should be done by forking that repo and following a standard pull request workflow. If you haven't looked at it already the docs are quite helpful:

https://conda-forge.org/docs/

If you run into any difficulties please let us know.

**Comment by mikofski** [comment]
Yay! Thanks!

**Pull request merged**